Content.Shared/Damage/Components/DamageOnPickupComponent.cs

```diff
@@ -0,0 +1,10 @@
+using Content.Shared.Damage.Systems;
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Damage.Components;
+
+[RegisterComponent, NetworkedComponent]
+public sealed partial class DamageOnPickupComponent : Component
+{
+
+}
```

Comment:
It is probably fine if this is combined with the ThornyComponent, just make the throwing optional via a bool DataField